component/newstorage/mem/mem.go

```diff
@@ -354,6 +354,7 @@ type memIterator struct {
 // Next moves the pointer to the next entry in the iterator. It returns false if the iterator is exhausted.
 func (m *memIterator) Next() (bool, error) {
 	if len(m.dbEntries) == m.currentIndex || len(m.dbEntries) == 0 {
+		m.dbEntries = nil
```

Comment:
could be potential race condition situation

Reply:
Shouldn't the iterator be 'owned' by the caller of 'Query'? In which case the iterator doesn't need to be thread-safe.

Reply:
Should be fixed now... let me know if you see any other issues

Reply:
> Shouldn't the iterator be 'owned' by the caller of 'Query'? In which case the iterator doesn't need to be thread-safe.

I would think so as well... while technically possible to create the problematic situation, I don't see how it would happen normally.